source/base/parameter_handler.cc

```diff
@@ -736,6 +848,102 @@ namespace Patterns
       return nullptr;
   }
 
+// I don't know how to do this more efficiently...
+  namespace
+  {
+
+    template<class T>
+    bool check_and_convert_to_any(const std::vector<std::string> &sv,
+                                  const PatternBase &t_pattern,
+                                  boost::any &v)
+    {
+      if (v.type() == typeid(std::vector<T> *))
+        {
+          std::vector<T> &vector = *boost::any_cast<std::vector<T> *>(v);
+          vector.resize(sv.size());
+          auto it = sv.begin();
+          for (auto &value : vector)
+            {
+              t_pattern.to_value(*(it++), value);
+            }
+          return true;
+        }
+      else
+        return false;
+    }
+
+
+    template<class T>
+    bool check_and_convert_to_string(const boost::any &v,
+                                     const PatternBase &t_pattern,
+                                     std::vector<std::string> &sv)
+    {
+      if (v.type() == typeid(const std::vector<T> *))
+        {
+          const std::vector<T> &vector = *boost::any_cast<const std::vector<T> *>(v);
+          sv.resize(vector.size());
+          auto it = sv.begin();
+          for (auto &value : vector)
+            {
+              *(it++) = t_pattern.to_string(value);
+            }
+          return true;
+        }
+      else
+        return false;
+    }
+
+  }
+
+
+  void List::string_to_any(const std::string &s, boost::any &v) const
+  {
```

Comment:
@tamiko , no, I'm throwing exceptions all the times when things don't conform to what this specific pattern does.